lib/sender.js

```diff
@@ -51,10 +65,92 @@ class Sender extends Writable {
     if (this._options.debug) {
       console.log('%s', converted.trim());
     }
-    return this._socket.write(converted, error => this._handleError(error,
-      callback))
+    const writeRes = this._socket.write(
+      converted, error => this._handleError(error, callback));
+    return this._relp.enabled ? writeRes && this._relp.txno : writeRes;
+  }
+
+  resend(message, txno, callback) {
+    if (!this._relp.enabled) {
+      callback(new Error('resend only supported with a RELP sender'));
+      return false;
+    }
+    const converted = this._convert(message, txno)
+    if (this._options.debug) {
+      console.log('%s', converted.trim());
+    }
+    const writeRes = this._socket.write(
+      converted, error => this._handleError(error, callback));
+    return writeRes && txno;
+  }
+
+  sendClose(callback) {
+    if (!this._relp.enabled) {
+      callback(new Error('sendClose only supported with a RELP sender'));
+      return false;
+    }
+    this._relp.closeTxno = this._nextRelpTxno();
+    const converted = `${this._relp.closeTxno} close 3 bye\n`;
+    if (this._options.debug) {
+      console.log('%s', converted.trim());
+    }
+    const writeRes = this._socket.write(
+      converted, error => this._handleError(error, callback));
+    return writeRes && this._relp.closeTxno;
+  }
+
+  _sendRelpOpen() {
+    const message = `relp_version=0
+relp_software=@devo/nodejs-sdk,${version},http://devo.com
+commands=syslog`;
+    this._relp.openTxno = this._nextRelpTxno();
+    const converted =
+      `${this._relp.openTxno} open ${message.length} ${message}\n`;
+    if (this._options.debug) {
+      console.log('%s', converted.trim());
+    }
+    this._socket.write(converted);
+  }
+
+  _onData(data) {
+    // pre: this._relp.enabled === true
+    this._relp.buffer += data.toString();
+    this._extractRsps();
+  }
+
+  _extractRsps() {
+    let m;
+    while((m = this._relp.buffer.match(RELP_RSP_REGEX))) {
```

Comment:
```suggestion
    while((let m = this._relp.buffer.match(RELP_RSP_REGEX)) !==  null) {
```
and then remove the `let m;` declaration above

Reply:
Ok to adding the strict equality, but I'm afraid JS does not accept declaration within a `while` conditional statement. Maybe you're confusing it with the `for` initialization block?

```js
// this is ok
for (let x = Math.random(); x < 0.8; x = Math.random()) { console.log('nope'); }
// this throws a syntax error
while ((let x = Math.random()) < 0.8) { console.log('nope'); }
// this throws too
if ((let x = Math.random()) < 0.8) { console.log('nope'); }
```

4cfef09348f6a1d8fbd12994a5b8a24145ac819a